### PR TITLE
fix: fix CoreAPI bug

### PR DESCRIPTION
**Pull request by captt-g**
I find when in rbac_with_domain , use the getAllAction API will come some bug. And I add some code to fix it .

### CODE REVIEW

File: Sources/Casbin/APi/ManagementApi.swift [hunk 1]
```diff
@@ -106,13 +106,13 @@ public extension CoreApi {
...
-        model.getValuesForFieldInPolicy(sec: "p", ptype: ptype, fieldIndex: 0)
+        model.getValuesForFieldInPolicy(sec: "p", ptype: ptype, fieldIndex: model.getModel()["p"]?["p"]?.tokens.firstIndex(of: "p_sub") ?? 0)
     }
     func getAllNamedObjects(ptype:String) -> [String] {
-        model.getValuesForFieldInPolicy(sec: "p", ptype: ptype, fieldIndex: 1)
+        model.getValuesForFieldInPolicy(sec: "p", ptype: ptype, fieldIndex: model.getModel()["p"]?["p"]?.tokens.firstIndex(of: "p_obj") ?? 1)
     }
     func getAllNamedActions(ptype:String) -> [String] {
-        model.getValuesForFieldInPolicy(sec: "p", ptype: ptype, fieldIndex: 2)
+        model.getValuesForFieldInPolicy(sec: "p", ptype: ptype, fieldIndex: model.getModel()["p"]?["p"]?.tokens.firstIndex(of: "p_act") ?? 2)
```
Comment (xiaobeiswift) [inline Sources/Casbin/APi/ManagementApi.swift:115]: it's nice!
Please modify the DefaultModel.getValuesForFieldInPolicy(sec:String,ptype:String,fieldIndex:Int)  function at the same time.
```swift
      public func getValuesForFieldInPolicy(sec:String,ptype:String,fieldIndex:Int) -> [String]  {
            let policy = getPolicy(sec: sec, ptype: ptype).reduce(Set<String>()) { acc, x in
            var acc = acc
            acc.insert(x[fieldIndex])
            return acc
        }
        return Array(policy)
    }
```

File: Tests/CasbinTests/RbacApiTests.swift [hunk 2]
```diff
@@ -42,7 +42,16 @@ final class RbacApiTests: XCTestCase {
...
+        XCTAssertEqual(["data1", "data2", "data2", "data2"], e.getAllObjects())
+        XCTAssertEqual(["alice", "bob", "data2_admin", "data2_admin"], e.getAllSubjects())
+        XCTAssertEqual(["data2_admin", "data1_admin"], e.getAllRoles())
+    }
+    func testCoreApi_for_RoleApi_with_domain() throws {
+        let e = try makeEnforer("examples/rbac_with_domains_model.conf", "examples/rbac_with_domains_policy.csv")
+        XCTAssertEqual(["read", "write", "read", "write"], e.getAllActions())
+        XCTAssertEqual(["data1", "data1", "data2", "data2"], e.getAllObjects())
+        XCTAssertEqual(["admin", "admin", "admin", "admin"], e.getAllSubjects())
+        XCTAssertEqual(["admin", "admin"], e.getAllRoles())
```
Comment (xiaobeiswift) [inline Tests/CasbinTests/RbacApiTests.swift:55]: Please remove duplicate elements,

ex:
```swift
     XCTAssertEqual(["read", "write"], e.getAllActions().sorted())
     XCTAssertEqual(["data1","data2"], e.getAllObjects().sorted())
```

File: Sources/Casbin/Model/DefaultModel.swift [hunk 3]
```diff
@@ -204,13 +204,15 @@ extension DefaultModel:Model {
...
         getPolicy(sec: sec, ptype: ptype).contains(rule)
     }
-    public func getValuesForFieldInPolicy(sec:String,ptype:String,fieldIndex:Int) -> [String] {
-        getPolicy(sec: sec, ptype: ptype).reduce([]) { acc, x in
-            var acc = acc
-            acc.append(x[fieldIndex])
-            return acc
-        }
-    }
+    public func getValuesForFieldInPolicy(sec:String,ptype:String,fieldIndex:Int) -> [String]  {
```
Comment (gaojian1123) [inline Sources/Casbin/Model/DefaultModel.swift:207]: `reduce(into:_:)` may be more suitable here.

```swift
let policy = getPolicy(sec: sec, ptype: type).reduce(into: Set<String>()) { acc, x in
    acc.insert(x[fieldIndex])
}
```

Comment (captt-g) [reply Sources/Casbin/Model/DefaultModel.swift:207]: e, I don't think this is necessary, but thank you.

File: Sources/Casbin/Model/DefaultModel.swift [hunk 4]
```diff
@@ -204,13 +204,15 @@ extension DefaultModel:Model {
...
+          let policy = getPolicy(sec: sec, ptype: ptype).reduce(Set<String>()) { acc, x in
+          var acc = acc
+          acc.insert(x[fieldIndex])
+          return acc
+      }
```
Comment (gaojian1123) [inline Sources/Casbin/Model/DefaultModel.swift:212]: Not only lines of code that `reduce(into:_:)` can reduce, it also can prevent copying set during each iteration. You know, when set is large, the copying operation will be time consuming. Although the set here may be quite small, I insist a method can avoid unnecessary copying should be better.

```suggestion
      let policy = getPolicy(sec: sec, ptype: ptype).reduce(into: Set<String>()) { acc, x in
          acc.insert(x[fieldIndex])
      }
```

Comment (captt-g) [reply Sources/Casbin/Model/DefaultModel.swift:212]: Thanks,I will take it to the code.

**Comment by hsluoyz** [comment]
@xiaobeiswift plz review.

File: Sources/Casbin/Model/DefaultModel.swift (hunk 4 above)
Comment (xiaobeiswift) [reply Sources/Casbin/Model/DefaultModel.swift]: This kind of writing should be reported. acc  is a 'let' constant

**Comment by xiaobeiswift** [comment]
@hsluoyz plz merge

**Pull request merged**